openmessaging-rocketmq/src/main/java/io/openmessaging/rocketmq/RocketMQClient.java

```diff
@@ -0,0 +1,109 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.openmessaging.rocketmq;
+
+import io.openmessaging.api.Client;
+import io.openmessaging.api.ClientConfiguration;
+import io.openmessaging.api.MessageId;
+import io.openmessaging.api.consumer.Consumer;
+import io.openmessaging.api.consumer.ConsumerConfiguration;
+import io.openmessaging.api.producer.Producer;
+import io.openmessaging.api.producer.ProducerConfiguration;
+import io.openmessaging.api.reader.Reader;
+import io.openmessaging.api.reader.ReaderConfiguration;
+import io.openmessaging.exception.OMSException;
+import io.openmessaging.internal.AccessPointURI;
+import java.util.concurrent.CompletableFuture;
+import org.apache.rocketmq.client.exception.MQClientException;
+import org.apache.rocketmq.client.producer.DefaultMQProducer;
+
+class RocketMQClient implements Client {
+
+    private final AccessPointURI uri;
+    private final ClientConfiguration conf;
+    private final DefaultMQProducer producer;
+
+    RocketMQClient(String url) {
+        this(url, new ClientConfiguration());
+    }
+
+    RocketMQClient(String url, ClientConfiguration conf) {
+        this.uri = new AccessPointURI(url);
+        this.conf = conf;
+
+        String producerGroupName = conf.getProperty("rocketmq.producer.group.name");
+        this.producer = new DefaultMQProducer(producerGroupName);
+    }
+
+    @Override
+    public Producer createProducer(String topic,
+                                   ProducerConfiguration conf)
+            throws OMSException {
+        try {
+            return new RocketMQProducer(producer, topic);
+        } catch (MQClientException e) {
+            throw new OMSException("create error", e);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Producer> createProducerAsync(String topic, ProducerConfiguration conf) {
+        return null;
+    }
+
+    @Override
+    public Consumer subscribe(String topic, String subscription, ConsumerConfiguration conf) throws OMSException {
+        return null;
+    }
+
+    @Override
+    public CompletableFuture<Consumer> subscribeAsync(String topic, String subscription, ConsumerConfiguration conf) {
+        return null;
+    }
+
+    @Override
+    public Reader createReader(String topic, MessageId startMessageId, ReaderConfiguration conf) throws OMSException {
+        return null;
+    }
+
+    @Override
+    public CompletableFuture<Reader> createReaderAsync(String topic, MessageId startMessageId, ReaderConfiguration conf) {
+        return null;
+    }
+
+    @Override
+    public CompletableFuture<Void> closeAsync() {
+        try {
+            close();
```

Comment:
This sync/async could be implemented by issuing `close()` from a background executor thread